laurat007/assignment1/Q2.java

```diff
@@ -0,0 +1,95 @@
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author laura
+ * The following program will get the K-th to last element in a linked list.
+ * It reads the input from a file containing on the first line a number N meaning the length of the list,
+ * and on the second line N numbers meaning the elements of the list.
+ */
+public class Q2 {
+	
+	public static final String INPUT_FILE = "in";
+	
+	static class MyList {
+		Object obj;
+		MyList next;
+		
+		void addLast(Object n) {
+			if(this.obj == null){
+				this.obj = n;
+				return;
+			}
+			
+			MyList aux, newNode;
+			newNode = new MyList();
+			newNode.obj = n;
+			newNode.next = null;
+			
+			aux = this;
+			while(aux.next != null) {
+				aux = aux.next;
+			}
+			aux.next = newNode;
+		}
+		
+		public String toString() {
+			String result = "";
+			MyList aux = this;
+			while(aux.next != null) { 
+				result += aux.obj + ", ";
+				aux = aux.next;
+			}
+			return result + aux.obj;
+		}
+	}
+	
+	public static void main(String[] args) {
```

Comment:
Can you also define a getKthToLast method and again call it from the main function after parsing the input file?
Once again this approach would allow you to write test that would just call the getKthToLast method with some specific example values and checking if it behaves as expected.

Reply:
Done new method